Project/searchResult.js

```diff
@@ -0,0 +1,310 @@
+            const data2 = await response2.json()
+            return data2
+        }
+
+        // Create an array to put in results of the full company detail
+        let results = [];
+
+        const createArrayWithFullData = async () => {
+            for (const key of data) {
+                const result = await fetchFullCompanyDetail(key.symbol)
+                results.push(result)
+            }
+
+            return results
+        }
+
+        // Using a method to connect the two datasets together, namely the "stock data" and "fullcompanydata". This is to create one merged dataset by using a unique identifier of Symbol
+
+        await createArrayWithFullData()
+        let mergedData = [];
+        for (const item of data) {
+            let match = await results.find(x => x.symbol === item.symbol);
+            if (match) {
+                mergedData.push({ ...item, ...match });
+            }
+        }
+
+        // Create a list in the html with the merged data
+        this.createList(mergedData)
+
+        // Disable the loading spinner
+        this.disableSpinner()
+
+
+
+    }
+
+    // Create a list with the returned object
+
+    async createList(data) {
+
+        let searchInput = searchValue.value
+
+        const list = await data.map(item => {
+            return `<li class="list"> <img class="image" src=${item.profile.image} alt="Picture Dosen't exist"> <a href="company.html?symbol=${item.symbol}"> ${item.name} <span>(${item.symbol})</span> <span id="percentage">(${this.convertToPercentages(item.profile.changesPercentage)})</span> </a> <button class="compare btn btn-info btn-sm text-center" id="${item.symbol}">Compare</button> </li>`;
+        }).join("");
+
+
+        // Create a regex that searches for the name and ticker and adds a space around them
+
+        const regex = new RegExp(`(?<=\\s|\\()${searchInput}`, "gi");
+        const outputString = `${list.replace(regex, (match) => `<span class="yellow">${match}</span>`)}`;
+
+        // Add the output to the container inner html
+
+        listContainer.innerHTML = `${outputString}`
+        listContainer.classList.add('list-group')
+
+        // Change the color of the percentages
+        this.changeColorPercentage()
+        this.compareButton = document.querySelectorAll("button")
+
+        // Call the event listener and run function
+        this.attachListeners(data)
+
+        const comparisons = new CompareStocks()
+
+
+    }
+
+    changeColorPercentage() {
+        let percentageCheck = document.querySelectorAll('#percentage')
+        for (let i = 0; i < percentageCheck.length; i++) {
+            if (percentageCheck[i].innerHTML.length === 7) {
+                percentageCheck[i].classList.add('green')
+            } else {
+                percentageCheck[i].classList.add('red')
+            }
+        }
+    }
+
+    // Function to convert to percentage
+    convertToPercentages(number) {
```

Comment:
is that method a doing something specif to this class, or maybe something more general, that other function/classes/code block may use? if that's the case, this code should be in other file inside another funcion.